File: sharedUI/src/commonMain/kotlin/br/com/ajudafio/presentation/book/list/BookListState.kt
package br.com.ajudafio.presentation.book.list

import br.com.ajudafio.book.domain.model.Book
import br.com.ajudafio.presentation.UiText

data class BookListState(
    val searchQuery: String = "",
    val searchResults: List<Book> = emptyList(),
    val favoriteBooks: List<Book> = emptyList(),
    val isLoading: Boolean = false,
    val selectedTabIndex: Int = 0,
    val errorMessage: UiText? = null
)

File: iosApp/iosApp/Book/Presentation/List/BookListState.swift
import SharedLogic

// Espelha br.com.ajudafio.presentation.book.list.BookListState do sharedUI.
struct BookListState {
    var searchQuery: String = ""
    var searchResults: [Book] = []
    var favoriteBooks: [Book] = []
    var isLoading: Bool = false
    var selectedTabIndex: Int = 0
    var errorMessage: String?
}
